src/plugins/map/MapObject.cpp

```diff
@@ -968,8 +967,10 @@ void MapObjects_Clear()
 {
 	GroundItemMap.clear();
 	SpawnMap.clear();
-	sLocationsMap.clear();
-	sMapLocs.clear();
+	for (auto maplocTemplate : gMapLocTemplates)
+	{
+		maplocTemplate->ClearReferenceToMapLoc();
```

Comment:
this should be in the destructor of MapLoc instead, then you don't need to call it explicitly.

Reply:
finding the template from the maploc can throw (not really).  If I try / catch that though, it means this might not get reset.

Reply:
```
MapObjectMapLoc::~MapObjectMapLoc()
{
	try
	{
		auto thisTemplate = GetMapLocByTag(m_tag);
		thisTemplate->ClearReferenceToMapLoc();
	}
	catch (...)
	{
	}
	RemoveMapLoc();
}
```

Reply:
why would it throw?

Reply:
my mistake, was tired.  This can be nullptr, not throw.  I'll fix heh